src/main/python/mlsearch/api_requester.py

```diff
@@ -312,23 +331,37 @@ def fetch_data(self) -> json:
                 self._fetch_paperwithcode()
 
             if self.params.get("source", "") == "github":
-                self._fetch_github()
+                try:
+                    self._fetch_github()
+                except BadCredentialsException:
+                    self.data["response_code"] = 400
+                    self.data["content"] = "Invalid Github developer key."
+                except RateLimitExceededException:
+                    self.data["response_code"] = 503
+                    self.data["content"] = "Access rate limitation reached."
 
             if self.params.get("source", "") == "youtube":
                 if not self._config.YOUTUBE_ORDER in self._config.VALID_YOUTUBE_ORDER:
                     self.data["response_code"] = 400
                     self.data["content"] = "Invalid Youtube Query Order."
                     return self.data
-
-                self._fetch_youtube(self.params.get("y_next_page_token", None))
+                try:
+                    self._fetch_youtube(self.params.get("y_next_page_token", None))
+                except HttpError as ex:
+                    print(str(ex))
```

Comment:
Maybe we can remove this log to? 🤔

Reply:
The purpose of print is to log the real root caused into cloud watch as we overwrite the real root caused before responding to the client. Google API did not provide a convenient ways to handle the exception rather than throwing HTTP Error with specific response code. I also cannot find the pre defined response code regarding to YouTube API too.

Reply:
The purpose of print is to log the real root caused into cloud watch as we overwrite the real root caused before responding to the client. Google API did not provide a convenient ways to handle the exception rather than throwing HTTP Error with specific response code. I also cannot find the pre defined response code regarding to YouTube API too.